### PR TITLE
Endpoints connection

### PR DESCRIPTION
**Pull request by AlbertMartinezM**
Changelog:
- When joining to a group, if you change username, it actually changes now.
- Plan members without username have "Unknown name" in their user cards.
- Add tag to user card to indicate whether a user is Admin or not
- Admin can remove any plan member, even themselves.
- When pressing "leave" "remove" or "join" button, the plan data is automatically updated

Also ran "npm run format" and merged all the conflicts (There were so many, hope have not screwed it up)

Just one thing:
The person who creates the plan can't set their username (so, it will appear as "Unknown user"), but the other members will have this option

### CODE REVIEW

File: src/components/user/UserCard.jsx [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+    planId,
+    adminId,
+    participants,
+    setParticipants,
+}) => {
+    const leavePlan = () => {
+        if (userUuid === currentUserUuid || currentUserUuid === adminId) {
+            deleteParticipant(planId, userUuid).then((r) => {
+                toast.success("Participant deleted");
+                // Ejemplo: Eliminar la fila con id igual a 2
```
Comment (jorbush) [inline src/components/user/UserCard.jsx:19]: Please remove this comment or translate if you think it's relevant.

File: src/components/user/UserCard.jsx [hunk 2]
```diff
@@ -1,7 +1,33 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (jorbush) [inline src/components/user/UserCard.jsx:1]: I think you don't use this new import

**Review by jorbush — CHANGES_REQUESTED** [review]
It looks nice! Some comments

**Review by jorbush — APPROVED** [review]
LGTM 😼 Nice work!!

File: src/components/user/UserCard.jsx (hunk 1 above)
Comment (AlbertMartinezM) [reply src/components/user/UserCard.jsx]: well, this is the language of Cristo nuestro señor, but will remove it for you, no worries

File: src/components/user/UserCard.jsx (hunk 2 above)
Comment (AlbertMartinezM) [reply src/components/user/UserCard.jsx]: oops my bad

**Review by jorbush — CHANGES_REQUESTED** [review]
In the description you have post: 
`The person who creates the plan can't set their username (so, it will appear as "Unknown user"), but the other members will have this option`
The thing is that the author of the plan have a field in the form called `Author` and you have to call `updateUsername()` when creating a new plan.

**Pull request merged**